File: Communication/FragComNavVModelDemo/app/src/main/java/edu/cs4730/fragcomnavvmodeldemo/DataViewModel.java
package edu.cs4730.fragcomnavvmodeldemo;

import android.app.Application;
import android.util.Log;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

public class DataViewModel extends AndroidViewModel {
    int num_one = 0;
    int num_two = 0;
    private MutableLiveData<String> item;

    public DataViewModel(@NonNull Application application) {
        super(application);
        item = new MutableLiveData<String>();
        item.setValue("default value");
    }

    public LiveData<String> getData() {
        return item;
    }

    String getItem() {
        return item.getValue();
    }

    void setItem(String n) {
        Log.d("VM", "data is " + n);
        item.setValue(n);
    }


}

File: Communication/FragComNavVModelDemo_kt/app/src/main/java/edu/cs4730/fragcomnavvmodeldemo_kt/DataViewModel.kt
package edu.cs4730.fragcomnavvmodeldemo_kt

import android.app.Application
import android.util.Log
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.LiveData

class DataViewModel(application: Application) : AndroidViewModel(application) {
    @JvmField
    var num_one = 0
    @JvmField
    var num_two = 0
    private val item: MutableLiveData<String> = MutableLiveData()
    val data: LiveData<String>
        get() = item

    fun getItem(): String? {
        return item.value
    }

    fun setItem(n: String) {
        Log.d("VM", "data is $n")
        item.value = n
    }

    init {
        item.value = "default value"
    }
}